flocker/provision/_install.py

```diff
@@ -800,6 +802,57 @@ def task_open_control_firewall(distribution):
     Open the firewall for flocker-control.
     """
     if is_centos(distribution):
+        open_firewall = open_firewalld
+    elif distribution == 'ubuntu-14.04':
+        open_firewall = open_ufw
+    else:
+        raise DistributionNotSupported(distribution=distribution)
+    return sequence([
+        open_firewall(service)
+        for service in ['flocker-control-api', 'flocker-control-agent']
+    ])
+
+
+def catch_exit_code(expected_exit_code):
+    """
+    :param int expected_exit_code: The expected exit code of the process.
+    :returns: An error handler function which accepts a 3-tuple(exception_type,
+        exception, None) and re-raises ``exception`` if the exit code is not
+        ``expected_exit_code`.
+    """
+    def error_handler(result):
+        exception_type, exception = result[:2]
+        if((exception_type is not ProcessTerminated) or
+           (exception.exitCode != expected_exit_code)):
+            raise exception
+    return error_handler
+
+
+def if_firewall_available(distribution, commands):
+    """
+    Open the firewall for remote access to control service if firewall command
+    is available.
+    """
+    if is_centos(distribution):
+        firewall_command = b'firewall-cmd'
+    elif distribution == 'ubuntu-14.04':
+        firewall_command = b'ufw'
+    else:
+        raise DistributionNotSupported(distribution=distribution)
+
+    # Only run the commands if the firewall command is available.
+    return run_from_args([b'which', firewall_command]).on(
+        success=lambda result: commands,
+        # XXX: Is this the exit code of the SSH command or the remote command?
```

Comment:
:grey_question: This seems like a relatively easy thing to find out by experimentation. I'll leave it up to you whether it's worth doing so or not before landing.